ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetails.tsx

```diff
@@ -0,0 +1,74 @@
+import React, { useCallback } from 'react';
+import { useParams } from 'react-router-dom';
+import {
+    Alert,
+    Breadcrumb,
+    BreadcrumbItem,
+    Divider,
+    PageSection,
+    Skeleton,
+} from '@patternfly/react-core';
+
+import BreadcrumbItemLink from 'Components/BreadcrumbItemLink';
+import PageTitle from 'Components/PageTitle';
+import useRestQuery from 'hooks/useRestQuery';
+import { complianceEnhancedCoveragePath } from 'routePaths';
+import { getSingleClusterCombinedStats } from 'services/ComplianceEnhancedService';
+import { getAxiosErrorMessage } from 'utils/responseErrorUtils';
+
+import ClusterDetailsHeader from './ClusterDetailsHeader';
+
+function ClusterDetails() {
+    const { clusterId } = useParams();
+
+    const listQuery = useCallback(() => getSingleClusterCombinedStats(clusterId), [clusterId]);
+    const { data: clusterStats, loading: isLoadingClusterInfo, error } = useRestQuery(listQuery);
+
+    const renderClusterNameBreadcrumb = () => {
+        if (error) {
+            return null;
+        }
+
+        return (
+            <BreadcrumbItem isActive>
+                {isLoadingClusterInfo ? (
+                    <Skeleton screenreaderText="Loading cluster name" width="150px" />
+                ) : (
+                    clusterStats?.cluster.clusterName
+                )}
+            </BreadcrumbItem>
+        );
+    };
+
+    return (
+        <>
+            <PageTitle title="Compliance Cluster Details" />
+            <PageSection variant="light" className="pf-u-py-md">
+                <Breadcrumb>
+                    <BreadcrumbItemLink to={`${complianceEnhancedCoveragePath}?tableView=Clusters`}>
+                        Clusters
+                    </BreadcrumbItemLink>
+                    {renderClusterNameBreadcrumb()}
+                </Breadcrumb>
+            </PageSection>
+            <Divider component="div" />
+            {error || clusterStats === null ? (
```

Comment:
The `|| clusterStats === null` caused me to pause long enough to ask whether `isLoadingClusterInfo` is relevant here as above to render a spinner? Although the cost is a nested ternary, error, loading, or data is such a common pattern. A slight variation that I will mention, not necessarily recommend, is `isLoadingClusterInfo || clusterStats === undefined` so TypeScript knows the value cannot be nullish in `clusterStats={clusterStats}` below and in similar future prop when you add the table.

Reply:
Yeah this is weird for sure. I reorganized it a bit in a follow up PR [here](ui/apps/platform/src/Containers/ComplianceEnhanced/ClusterCompliance/Coverage/Clusters/ClusterDetails.tsx), but I think I like your idea of checking for undefined as opposed to !null